### ISSUE TITLE
Terminology Dictionary

### ISSUE THREAD

**Issue by sustained** [vuejs-sv/vuejs.org#4]
If you know a translation that I'm missing (see unchecked items below) or have a better translation than the one I'm using then please leave a comment.

I don't have very much experience with strictly _technical_ reading/writing in this language, hence the need for this little dictionary.

# Technical Dictionary

## No Translation Required

**NOTE:** Recommended to translate upon first occurrence but not afterwards.

| Swedish (First Time Occurrence) | English |
| - | - |
| computed (beräknad variabel) | computed |
| event (händelse) | event |
| hook (krok) | hook |
| lifecycle hook (livscykelkrok) | lifecycle hook |
| prop (egenskap) | prop |
| state (tillstånd) | state |
| template (mall) | template |
| watcher (bevakare) | watcher |

## Nouns

| Swedish | English |
| - | - |

| en abstraktion | an abstraction |
| ett användargränssnitt | a user interface |
| ett attribut | an attribute |
| ett bibliotek | a library |
| en bindning | a binding |
| en cache | a cache |
| ett direktiv | a directive |
| en egenskap | a property |
| (ett) element | an element |
| en flik | a tab |
| en frontendutvecklare | a front-end developer |
| en funktion | a feature; a function |
| ett fält | a field |
| (en) inmatning | (an) input |
| en instans | an instance |
| en klass | a class |
| (en) kod | (a) code |
| en kommandorad | a command-line interpeter |
| en komponent | a component |
| en konstruktor | a constructor |
| en kryssruta | a checkbox |
| (en) logik | (a) logic |
| en mall | a template |
| en metod | a method |
| en operatör | an operator |
| en parameter | a parameter |
| (en) prestanda | (a) performance |
| ett ramverk | a framework |
| en scope; (en räckvidd) | a scope |
| en syntax; (en satslära) | a syntax |
| en styrstruktur | a control structure |
| en slinga | a loop |
| ett tillstånd | a state |
| en transition | a transition |
| ett uttryck | an expression |
| en variabel | a variable |
| en verktyg | a tool; (a) tooling |
| ett värde | a value |
| ett repo | a repository (i.e. SCM) |
| ett villkor | a conditional |
| (en) verktyg | (a) tooling |

## Adjectives

| Swedish | English |
| - | - |
| beräknad | computed |
| deklarativ | declarative |
| interaktiv | interactive |
| progressiv | progressive |
| reaktiv | reactive |
| responsiv | responsive |

## Verbs

| Swedish | English |
| - | - |
| anropa; (kalla) | to call, invoke (i.e. a function) |
| binda | to bind |
| exekvera | to execute |
| generera | to generate |
| lagra | to store |
| rendera | to render |
| växla; (byta) | to toggle |
| ärva | to inherit |

## Phrases

| Swedish | English |
| - | - |
| ett öppen källkodsprojekt | an open-source project |

## Hard to translate

| Swedish | English |
| - | - |
| - | to apply (e.g. changes none of the Swedish words really work) |

**Comment by sustained** [comment]
How on earth do we properly translate "an open source project"?

* ett öppen källkod projekt?
* ett öppen/t källkod-projekt? :thinking: 
* ett öppet källprojekt?

Who even knows... :frowning:

**Comment by sustained** [comment]
How do we translate "repository" as in, you know, source control management?

**Comment by Pwntus** [comment]
## Nouns
`(en) cache` - (a) cache
`ett repository`; (`ett repo`, `en kodsamling`) - a repository
_No good translation, repository/repo should be good._

`(ett) villkor` - a conditional
`flera villkor` - multiple conditions
`villkoret` - the condition
`verktyg` - (a) tooling

## Verbs
`växla`; (`byta`) - to toggle

## Phrases
`ett öppen källkodsprojekt`

**Comment by sustained** [comment]
Changes incorporated, thanks!

**Comment by sustained** [comment]
Incorporated changes from https://pastebin.com/egGdebv1 per request.

**Comment by sustained** [comment]
Some more trouble words:

* :question: (ett) element - an element
* :question: (en) direktiv - a directive
* :question: (en) transition - a transition
* :question: att applicera - to apply

I know that gälla is used for the sense of like "this applies only to ..." and tillämpa is sort of like "use in practice" and applicera is like to apply as in "to the skin" as well as some other meanings and so on. But in a programming context? Like to "apply a transition effect" for example? I have no idea. Seems safest to go with applicera, so that's what I have been doing.

**Comment by sustained** [comment]
Array can be translated as fält, uppställning or tabell but since we use fält for field I think we should stick to tabell?

**Comment by Pwntus** [comment]
The only change in your list is to use `ett` instead of `en` here:
(ett) direktiv - a directive

As for `applicera` in a programming context; it's OK to do as you say, but it could come of as a direct translation and not how you would naturally say it. In this case, I'd probably change the word, so `apply a transition effect` would be something like `skapa en övergångseffekt` (=create a transition effect).

Lastly, I think Array should be added to the "no translation required" list with an initial translation only once: `Array (fält)`.

**Comment by sustained** [comment]
See that's the thing, I would have thought that it'd make more sense to use the "no translation required" list for uncommon terms.

Basically I've been going off of "if I can find several Wikipedia articles relating to computer science that use the Swedish term then it's a valid term". And tabell and fält as far as I can tell are valid terms.

We definitely need to come to an agreement, in any case. It might be helpful to create some kind of script or I suppose git hook that checks that we are conforming to our conventions, even?

I can maybe look into that though I'm not sure how good of an idea it actually is. Especially considering the surface area we have to cover (i.e. not so much that doing it manually is a great hassle).

**Comment by sustained** [comment]
Some more I'm unsure about:

- to attach (context = an event listener)
- an event listener

It might actually be useful for me (I probably have more free time!) to do a sweep of the docs and make a list of all the terms that I think could be a problem. Then we can figure all of this out in one fell swoop and afterwards simply focus on the actual task at hand.

Another approach would be to just leave the English word/term/phrasing in for all of these kinds of occurences and then when we're done translating the rest we can compile a list, decide which should be translated, which shouldn't then basically find-and-replace the code- uhh I mean docsbase.

Thoughts?

**Comment by Pwntus** [comment]
I like the idea with a script :smile: 
Another idea is to make a temporary convention, similar to your suggestion with leaving the English word/term/phrasing, where we explicitly insert a "magic string" so that we easily can find them later (and maybe automatically replace them based on a terminology dictionary).

E.g.
```
Object or array defaults must be returned from a factory function.
```
:arrow_right:
```
Standardvärden som är objekt eller REPLACE[array, plural] måste returneras från en REPLACE[factory function, singular]
```

**Comment by Pwntus** [comment]
Essentially, insert the magic string ”REPLACE” _when in any doubt_. This results in a single CTRL+F action when later applying terms that eventually get agreed upon.

**Comment by sustained** [comment]
I quite like the convention idea but that becomes slightly more complicated when you take into account all possible conjugations... no?

Surely for it to be truly worthwhile (and for us to be able to __automatically__ replace all occurrences) we would need to do e.g. `REPLACE[array, nominative, indefinite, plural]`. and at that point it's taking up like half of the screen width. :joy: 

![image](https://user-images.githubusercontent.com/5995080/61191999-a1c0e280-a6a8-11e9-9a08-7f2738779997.png)

**Comment by sustained** [comment]
Sure, just adding a `(REPLACE)` is enough if we intend to go over things __manually__. I have actually been doing basically that on the thing I'm working on now (actually I was doing `(TRANS-REQ)`.

Okay, I think I'm just gonna add a `(TRANSLATE)` after anything I am unsure on for now then.

(The word translate is far less likely to be used in the docs than replace).

**Comment by sustained** [comment]
Then we can grep the entire "codebase" when we're "done". Sounds like a plan?

**Comment by Pwntus** [comment]
Yes, TRANSLATE it is. Yup, I just realized that automatically replacing is a lot more complicated as you say, and isn’t a realistic approach 😂

**Comment by sustained** [comment]
@Pwntus I have adopted this convention in the current document:

- `(TRANSLATE!)` - it was left as English/no translation attempt was made
- `(TRANSLATE?)` - a translation attempt was made but is awkward, or I'm unsure about it etc.

:+1:

**Comment by sustained** [comment]
Examples:

> I avsikt att låta användare interagerar med din app kan vi använda `v-on`-direktivet för att attach event listeners (TRANSLATE!) som anropar metoder på våra Vue-instanser:

> Vue ger också `v-model`-direktivet som gör det lätt som en plätt att få tvåvägsbindning (TRANSLATE?) mellan formulärsinmatningen (TRANSLATE?) och appen.

I think this is pretty good?

(Some of those translations are _soooo_ awkward, I know).

**Comment by Pwntus** [comment]
Yes this is even better. When possible, a separate issue should be made, explaining the convention for (possibly) other translators to see.
